subsys/net/lib/nrf_cloud/src/nrf_cloud_transport.c

```diff
@@ -608,6 +622,66 @@ static void aws_fota_cb_handler(struct aws_fota_event *fota_evt)
 }
 #endif /* defined(CONFIG_AWS_FOTA) */
 
+static int nct_settings_set(const char *key, size_t len_rd,
+			    settings_read_cb read_cb, void *cb_arg)
+{
+	if (!key) {
+		return -EINVAL;
+	}
+
+	int read_val;
+
+	LOG_DBG("Settings key: %s, size: %d", log_strdup(key), len_rd);
+
+	if (!strncmp(key, SETTINGS_KEY_PERSISTENT_SESSION,
+		     strlen(SETTINGS_KEY_PERSISTENT_SESSION)) &&
+	    (len_rd == sizeof(read_val))) {
+		if (read_cb(cb_arg, (void *)&read_val, len_rd) == len_rd) {
+#if defined(CONFIG_CLOUD_PERSISTENT_SESSIONS)
+			persistent_session = (bool)read_val;
+#endif
+			LOG_DBG("Read setting val: %d", read_val);
+			return 0;
+		}
+	}
+	return -ENOTSUP;
+}
+
+int save_session_state(const int session_valid)
+{
+	int ret = 0;
+
+#if defined(CONFIG_CLOUD_PERSISTENT_SESSIONS)
+	LOG_DBG("Setting session state: %d", session_valid);
+	persistent_session = (bool)session_valid;
+	ret = settings_save_one(SETTINGS_FULL_PERSISTENT_SESSION,
```

Comment:
I am wondering if it makes sense to not write the settings if they have not changed, in order to reduce unnecessary Flash wear?

Reply:
correct, there is no explicit check in this function, however the function only gets called when the session state is changing.
also `settings_fcb_save` does such a check before writing.